core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerBackend.scala

```diff
@@ -46,7 +47,7 @@ private[spark] class MesosSchedulerBackend(
   with MesosSchedulerUtils {
 
   // Which slave IDs we have executors on
-  val slaveIdsWithExecutors = new HashSet[String]
+  val slaveIdsWithExecutors = new HashMap[String, MesosExecutorInfo]
```

Comment:
this should be renamed `slaveIdToExecutorInfo`. Then you'll also need to update the comment above.